### PR TITLE
PLANNER-903: Made all Serializers/Deserializers internal to fix conflicts from different versions

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
@ge0ffrey This works with Wildfly 10.0.0.Final. I haven't tested on EAP 7 yet; should I test it on EAP 7.1.0 Beta or EAP 7.0.0?

### CODE REVIEW

**Comment by ge0ffrey** [comment]
@Christopher-Chianelli EAP 7.1.0 Beta is more important for us because we need to think long term and we need to get all our ducks in a row by the BRMS 7.0 release.

File: optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/jackson/LocalDateDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package org.optaplanner.openshift.employeerostering.shared.jackson;
```
Comment (ge0ffrey) [inline optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/jackson/LocalDateDeserializer.java]: Copy the copyright/license text from the original file, for legal purposes.

File: optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/jackson/LocalDateDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.JsonTokenId;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+public class LocalDateDeserializer extends JsonDeserializer<LocalDate> {
```
Comment (ge0ffrey) [inline optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/jackson/LocalDateDeserializer.java]: Add a comment that clearly indicates when this class would become obsolete.

By copying them, we don't get automatic bugfixes as we upgrade to newer jackson versions, but that is indeed the lesser evil today. This is a very good, practical PR.
However, that situation won't stay the case: at some point every modern WildFly/EAP version will contain the jackson java.time support out of the box and we no longer need to duplicate them.
The comment should say something like "// TODO remove me when our minimum WildFly version is x and the minimum EAP version is y"

File: pom.xml [hunk 3]
```diff
@@ -63,7 +63,7 @@
     <version.com.github.nmorel.gwtjackson>0.14.2</version.com.github.nmorel.gwtjackson>
     <version.com.github.nmorel.gwtjackson.restprocessor>0.4.1</version.com.github.nmorel.gwtjackson.restprocessor>
     <version.org.gwtbootstrap3>0.9.4</version.org.gwtbootstrap3>
-    <version.org.wildfly.wildfly-dist>10.1.0.Final</version.org.wildfly.wildfly-dist>
+    <version.org.wildfly.wildfly-dist>10.0.0.Final</version.org.wildfly.wildfly-dist>
```
Comment (ge0ffrey) [inline pom.xml]: we probably still want to push the latest WildfFly final version, despite also being compatible with 10.0

**Comment by Christopher-Chianelli** [comment]
@ge0ffrey I have added your suggested changes

**Pull request merged**